src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -90,7 +90,32 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		@IWorkspaceConfigurationService private _workspaceConfigurationService: IWorkspaceConfigurationService,
 		@IClipboardService private _clipboardService: IClipboardService,
 		@ICommandService private _commandService: ICommandService
-	) { }
+	) {
+		this.initializeConnectionProviders();
+	}
+
+	private initializeConnectionProviders() {
+		this.setConnectionProviders();
+		if (this._capabilitiesService) {
+			this._capabilitiesService.onCapabilitiesRegistered(() => {
```

Comment:
do we need to ensure this listener is disposed?

Reply:
as we discussed this service only 1 instance and will live across the lifetime of ADS, making it disposable and hook it up to make sure it is getting disposed doesn't seem to be simple task. leaving it as s.